src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/RemoveResourceDesignerStep.cs

```diff
@@ -0,0 +1,211 @@
+		CustomAttribute mainDesignerAttribute;
+		Dictionary<string, int> designerConstants;
+		Regex opCodeRegex = new Regex (@"([\w]+): ([\w]+) ([\w.]+) ([\w:./]+)");
+
+		protected override void Process ()
+		{
+			// resolve the MainAssembly Resource designer TypeDefinition
+			AndroidLinkConfiguration config = AndroidLinkConfiguration.GetInstance (Context);
+			if (config == null)
+				return;
+			foreach(var asm in config.Assemblies) {
+				if (FindResourceDesigner (asm, mainApplication: true, designer: out mainDesigner, designerAttribute: out mainDesignerAttribute)) {
+					mainAssembly = asm;
+				 	break;
+				}
+			}
+			if (mainDesigner == null) {
+				Context.LogMessage ($"  Main Designer not found.");
+				return;
+			}
+			Context.LogMessage ($"  Main Designer found {mainDesigner.FullName}.");
+			designerConstants = BuildResourceDesignerFieldLookup (mainDesigner);
+		}
+
+		protected override void EndProcess ()
+		{
+			if (mainDesigner != null) {
+				Context.LogMessage ($"  Setting Action on {mainAssembly.Name} to Save.");
+				Annotations.SetAction (mainAssembly, AssemblyAction.Save);
+			}
+		}
+
+		bool FindResourceDesigner (AssemblyDefinition assembly, bool mainApplication, out TypeDefinition designer, out CustomAttribute designerAttribute)
+		{
+			string designerFullName = null;
+			designer = null;
+			designerAttribute = null;
+			foreach (CustomAttribute attribute in assembly.CustomAttributes)
+			{
+				if (attribute.AttributeType.FullName == "Android.Runtime.ResourceDesignerAttribute")
+				{
+					designerAttribute = attribute;
+					if (attribute.HasProperties)
+					{
+						foreach (var p in attribute.Properties)
+						{
+							if (p.Name == "IsApplication" && (bool)p.Argument.Value == (mainApplication ? mainApplication : (bool)p.Argument.Value))
+							{
+								designerFullName = attribute.ConstructorArguments[0].Value.ToString ();
+								break;
+							}
+						}
+					}
+					break;
+
+				}
+			}
+			if (string.IsNullOrEmpty(designerFullName))
+				return false;
+
+			foreach (ModuleDefinition module in assembly.Modules)
+			{
+				foreach (TypeDefinition type in module.Types)
+				{
+					if (type.FullName == designerFullName)
+					{
+						designer = type;
+						return true;
+					}
+				}
+			}
+			return false;
+		}
+
+		Dictionary<string, int> BuildResourceDesignerFieldLookup (TypeDefinition type)
+		{
+			var output = new Dictionary<string, int> ();
+			foreach (TypeDefinition definition in type.NestedTypes)
+			{
+				foreach (FieldDefinition field in definition.Fields)
+				{
+					string key = $"{definition.Name}::{field.Name}";
+					if (!output.ContainsKey (key))
+						output.Add(key, int.Parse (field.Constant?.ToString () ?? "0"));
+				}
+			}
+			return output;
+		}
+
+		void ClearDesignerClass (TypeDefinition designer)
+		{
+			Context.LogMessage ($"    TryRemoving {designer.FullName}");
+			designer.NestedTypes.Clear ();
+			designer.Methods.Clear ();
```

Comment:
This will presumably remove the static constructor on the `Resource` type itself?